src/main/java/uk/gov/pay/ledger/agreement/service/AgreementService.java

```diff
@@ -111,7 +122,16 @@ public AgreementSearchResponse searchAgreements(AgreementSearchParams searchPara
                 .withPaginationBuilder(paginationBuilder);
     }
 
+    public List<EventDto> findEvents(String agreementExternalId, String serviceId) {
+        var agreementEntity = findAgreementEntity(agreementExternalId, true, null, serviceId)
```

Comment:
It would be good to test 404 exception too

Reply:
Agreed; done